neqo-transport/src/connection/mod.rs

```diff
@@ -350,15 +348,36 @@ impl Connection {
         );
         tps.set_empty(tparams::DISABLE_MIGRATION);
         tps.set_empty(tparams::GREASE_QUIC_BIT);
+    }
 
-        tps.set_integer(
+    /// Read connection parameters and update transport parameters.
+    fn read_parameters(&mut self) -> Res<()> {
```

Comment:
I am wondering, shouldn't `read_parameters` return `Res<TransportParameters>` and be responsible for creating `TransportParameters`, setting its default and reading `self.conn_params`?
I do not like that there are 2 steps and in between `TransportParameters` are not correct.

Reply:
That's good feedback, but it's a fairly large restructuring.  Opened #1073 for that.  This is definitely large enough to warrant follow-up.

Reply:
I agree, thanks.